python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -259,17 +263,49 @@ def _get_runs_to_dequeue(
             tag_concurrency_limits_counter = TagConcurrencyLimitsCounter(
                 tag_concurrency_limits, in_progress_runs
             )
-
             batch += queued_runs
             batch = self._priority_sort(batch)
 
+            if instance.run_coordinator.should_block_op_concurrency_limited_runs:
+                try:
+                    global_concurrency_limits_counter = GlobalOpConcurrencyLimitsCounter(
+                        instance,
+                        batch,
+                        in_progress_run_records,
+                        instance.run_coordinator.op_concurrency_slot_buffer,
+                    )
+                except:
+                    self._logger.error(
+                        "Failed to initialize op concurrency counter: \n%s",
+                        serializable_error_info_from_exc_info(sys.exc_info()),
+                    )
+                    # when we cannot initialize the global concurrency counter, we should fall back
+                    # to not blocking any runs based on op concurrency limits
+                    global_concurrency_limits_counter = None
+            else:
+                global_concurrency_limits_counter = None
+
             to_remove = []
             for run in batch:
                 if tag_concurrency_limits_counter.is_blocked(run):
                     to_remove.append(run)
                 else:
                     tag_concurrency_limits_counter.update_counters_with_launched_item(run)
 
+                if (
+                    global_concurrency_limits_counter
+                    and global_concurrency_limits_counter.is_blocked(run)
+                ):
+                    to_remove.append(run)
+                    concurrency_keys = (
+                        run.root_op_concurrency.key_counts.keys() if run.root_op_concurrency else []
+                    )
+                    self._logger.info(
+                        f"Keeping {run.run_id} queued due to blocked op concurrency keys: {concurrency_keys}"
+                    )
```

Comment:
(appreciating that i asked about adding this logging) is this likely to be pretty high volume? There are worlds I could imagine where this is logging hundreds or thousands of log lines every 5 seconds

We could make it debug level potentially?

Reply:
I think this is likely to be high volume... I'm inclined to drop it, tbh

Reply:
Particularly since its logging to the event log when its in this state, likely fine